metrics/tag.go

```diff
@@ -173,14 +176,15 @@ func init() {
 // unicode. Tags will be converted to lowercase."
 //
 // Note that this function does not impose the length restriction described above.
-func normalizeTag(in string) string {
+func normalizeTag(in string, validChars map[rune]struct{}) string {
+	var builder strings.Builder
+	builder.Grow(len(in))
 	in = strings.ToLower(in)
```

Comment:
if we do the lower-casing rune-by-rune can we avoid allocating another string?

Reply:
Stole this from the ToLower impl, but I think you can just add it as an else statement where we check for valid chars:
```go
		} else if 'A' <= r && r <= 'Z' {
			r += 'a' - 'A'
```